libr/bin/format/mach0/mach0.c

```diff
@@ -3151,6 +3151,13 @@ static void parse_relocation_info(struct MACH0_(obj_t) *bin, RSkipList *relocs,
 	}
 
 	ut64 total_size = num * sizeof (struct relocation_info);
+	if (offset > bin->size) {
+		return;
+	}
+	if (total_size > bin->size) {
```

Comment:
@trufae This definitely fixes the crash for security but I think there might be a way to make the codebase lighter.

Reply:
What do you mean by making the codebase lighter?

Reply:
I mean remove the if and let a single num allocation.

Reply:
The limit is the boundary of the file size you cant use a constant here